neurom/features/morphology.py

```diff
@@ -0,0 +1,286 @@
+        vector = morphmath.vector(section[0], soma.center)
+        norm_vector = np.linalg.norm(vector)
+
+        if norm_vector >= np.finfo(type(norm_vector)).eps:
+            return np.arcsin(vector[COLS.Y] / norm_vector)
+        raise ValueError("Norm of vector between soma center and section is almost zero.")
+
+    return [_elevation(n.root_node.points, morph.soma)
+            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+
+
+@feature(shape=(...,))
+def trunk_vectors(morph, neurite_type=NeuriteType.all):
+    """Calculates the vectors between all the trunks of the morphology and the soma center."""
+    return [morphmath.vector(n.root_node.points[0], morph.soma.center)
+            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+
+
+@feature(shape=(...,))
+def trunk_angles(morph, neurite_type=NeuriteType.all):
+    """Calculates the angles between all the trunks of the morph.
+
+    The angles are defined on the x-y plane and the trees
+    are sorted from the y axis and anticlock-wise.
+    """
+    vectors = np.array(trunk_vectors(morph, neurite_type=neurite_type))
+    # In order to avoid the failure of the process in case the neurite_type does not exist
+    if len(vectors) == 0:
+        return []
+
+    def _sort_angle(p1, p2):
+        """Angle between p1-p2 to sort vectors."""
+        ang1 = np.arctan2(*p1[::-1])
+        ang2 = np.arctan2(*p2[::-1])
+        return ang1 - ang2
+
+    # Sorting angles according to x-y plane
+    order = np.argsort(np.array([_sort_angle(i / np.linalg.norm(i), [0, 1])
+                                 for i in vectors[:, 0:2]]))
+
+    ordered_vectors = vectors[order][:, [COLS.X, COLS.Y]]
+
+    return [morphmath.angle_between_vectors(ordered_vectors[i], ordered_vectors[i - 1])
+            for i, _ in enumerate(ordered_vectors)]
+
+
+@feature(shape=(...,))
+def trunk_origin_radii(morph, neurite_type=NeuriteType.all):
+    """Radii of the trunk sections of neurites in a morph."""
+    return [n.root_node.points[0][COLS.R]
+            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+
+
+@feature(shape=(...,))
+def trunk_section_lengths(morph, neurite_type=NeuriteType.all):
+    """List of lengths of trunk sections of neurites in a morph."""
+    return [morphmath.section_length(n.root_node.points)
+            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+
+
+@feature(shape=())
+def number_of_neurites(morph, neurite_type=NeuriteType.all):
+    """Number of neurites in a morph."""
+    return sum(1 for _ in iter_neurites(morph, filt=is_type(neurite_type)))
+
+
+@feature(shape=(...,))
+def neurite_volume_density(morph, neurite_type=NeuriteType.all):
+    """Get volume density per neurite."""
+    return [nf.volume_density(n)
+            for n in iter_neurites(morph, filt=is_type(neurite_type))]
+
+
+def sholl_crossings(morph, center, radii, neurite_type=NeuriteType.all):
+    """Calculate crossings of neurites. Not a feature.
```

Comment:
well, it can be a long discussion. I don't know where to move. Lets keep it as is for now.

Reply:
It can be made a feature but for that `center` and `radii` must have default values. It is required because `morph_stats` module makes a dynamic config from all registered features. This config is called a full config, and can be used by users to get statistics with all features.

Reply:
Yeah, it's indeed hard to tell where these functions should go, especially if we don't want to change all the package structure. Let's keep them here for now and we might transform them into features later if it is needed.